subtype-parser/subtype.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python
+
+# Load required modules
+import os, sys
+
+def get_parser():
+    # Parse arguments
+    import argparse
+    description = 'Converts subtype and core event input data into'\
+                  'appropriate input for subtype-comet'
+    parser = argparse.ArgumentParser(description=description)
+
+    # Mutation data
+    parser.add_argument('-m', '--mutation_matrix', required=True,
+                        help='File name for mutation data.')
+    parser.add_argument('-sub', '--subtype', required=True,
+                        help='File with a list of subtype for performing subtype-comet.')
+    parser.add_argument('-ce', '--core_events', default=None,
+                        help='File with a list of core events for performing subtype-comet.')
+    parser.add_argument('-td', '--temp_directory', default='./',
+                        help='Directory for storing the temporary output of this conversion.')
+    return parser
+
+
+def run( args ):
+    # Parsing arguments
+    mutationMatrix = args.mutation_matrix
+    tempDirectory = args.temp_directory
+    subtypeList = args.subtype
+
+    # Parsing core events list, if available
+    if args.core_events:
+        with open(args.core_events) as f:
+            coreEvents = set( l.rstrip() for l in f )
+    else:
+        coreEvents = set()
+
+    # Parsing subtype information
+    from collections import defaultdict
+    subtypeDict = defaultdict( lambda: None, dict() )
+    subtypes = set()
+    with open( subtypeList ) as f:
+        sts = [ l.rstrip().split( '\t' ) for l in f if not l.startswith( '#' ) ]
+        for p, s in sts:
+            subtypeDict[p] = s
+            subtypes.add( s )
+
+    # Parsing mutation matrix
+    with open( mutationMatrix ) as f:
+        MM = [ l.rstrip().split( '\t' ) for l in f if not l.startswith( '#' ) ]
+
+    # Generating output for new mutation matrix with added subtype information
+    if tempDirectory[-1] != '/':
+        tempDirectory += '/'
+    with open( tempDirectory + 'temp.snv', "w" ) as out:
+        output = []
+        for p in MM:
+            if subtypeDict[p[0]] != None:
```

Comment:
This is not quite right. If patient _P_ is of subtype _T_, we want to add events (aberrations) for all subtypes **except** _T_. That way genes with alterations that are mutually exclusive with the subtype _T_ alteration are actually enriched for mutations in _T_.